x-pack/plugin/logsdb/src/main/java/org/elasticsearch/xpack/logsdb/SyntheticSourceIndexSettingsProvider.java

```diff
@@ -167,4 +172,29 @@ private IndexMetadata buildIndexMetadataForMapperService(
         tmpIndexMetadata.settings(finalResolvedSettings);
         return tmpIndexMetadata.build();
     }
+
+    /**
+     * The GA-ed use cases in which synthetic source usage is allowed with gold or platinum license.
+     */
+    boolean isLegacyLicensedUsageOfSyntheticSourceAllowed(IndexMode templateIndexMode, String indexName, String dataStreamName) {
+        if (templateIndexMode == IndexMode.TIME_SERIES) {
+            return true;
+        }
+
+        // To allow the following patterns: profiling-metrics and profiling-events
+        if (dataStreamName != null && dataStreamName.startsWith("profiling-")) {
+            return true;
+        }
+        // To allow the following patterns: .profiling-sq-executables, .profiling-sq-leafframes and .profiling-stacktraces
+        if (indexName.startsWith(".profiling-")) {
+            return true;
+        }
```

Comment:
@danielmitterdorfer Double checking, is the pattern sufficient in selecting the profiling data streams and indices that currently use synthetic source? Synthetic source will require an enterprise license from 8.17.0, but for profiling's synthetic source usage we will allow gold/platinum license for a little longer.